.github/workflows/summarize-checks.yaml

```
@@ -0,0 +1,55 @@
+name: "[TEST-IGNORE] Summarize Checks"
+
+on:
+  # We care about more labels than those that trigger a workflow run right? As in there are labels added that should update
+  # next steps to merge, but those labels might not trigger one of our status update runs.
+  # So we run this workflow on any label added to a PR.
+  workflow_run:
+    # todo, I feel like I'm missing a couple. I recall 6 being necessary/present.
+    workflows:
+    - "\\[TEST-IGNORE\\] Swagger SemanticValidation - Set Status"
+    - "\\[TEST-IGNORE\\] Swagger ModelValidation - Set Status"
+    - "\\[TEST-IGNORE\\] Swagger Avocado - Set Status"
+    - "Swagger LintDiff - Set Status"
+    - "SDK Validation Status"
+    types: [completed]
+  pull_request_target: # when a PR is labeled. NOT pull_request, because that would run on the PR branch, not the base branch.
+    types:
+      - labeled
+      - unlabeled
+
+permissions:
+  actions:       read   # to inspect workflow_run metadata
+  contents:      read   # for actions/checkout
+  checks:        read   # for octokit.rest.checks.listForRef
+  statuses:      read   # for octokit.rest.repos.getCombinedStatusForRef
+  issues:        write  # to post comments via the Issues API I'm not certain if I need this one or pull-requests only. we'll find out!
+  pull-requests: write  # to post comments via the Pull Requests API
+
+jobs:
+  run-summarize-checks:
+    name: "[TEST-IGNORE] Summarize Checks"
+    runs-on: ubuntu-24.04
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          # Only needs .github folder for automation, not the files in the PR (analyzed in a
+          # separate workflow).
+          sparse-checkout: |
+            .github
```

```suggestion
      # *** IMPORTANT ***
      # For workflows that are triggered by the pull_request_target event, the workflow runs in the
      # context of the base of the pull request.  You should make sure that you do not check out,
      # build, or run untrusted code from the head of the pull request.
      - uses: actions/checkout@v4
        with:
          # Only needs .github folder for automation, not the files in the PR (analyzed in a
          # separate workflow).
          #
          # Uses the .github folder from the PR base branch (pull_request_target trigger),
          # or the repo default branch (other triggers).
          sparse-checkout: |
            .github
```

NIT: Please copy this exact step including the warning comments.  It's very important for security in priviliged workflows.

---

Got it. Just the comment. Yeah no worries.

I read the step a couple times and was like

> Mike that's an exact clone